File: Mod07/Labfiles/Solution/VB/ClassTracker_VB/ClassTracker_VB/NonCreditCourse.vb
Public Class NonCreditCourse
    Inherits Course

    Public Function CalculateGrade(grade As Single) As Char

        Dim letterGrade As Char

        If grade >= 90 Then
            letterGrade = "A"c
        ElseIf (grade >= 80 And grade <= 89) Then
            letterGrade = "B"c
        ElseIf (grade >= 70 And grade <= 79) Then
            letterGrade = "C"c
        ElseIf (grade >= 66 And grade <= 69) Then
            letterGrade = "D"c
        Else
            letterGrade = "F"c
        End If

        Return letterGrade

    End Function

End Class

File: Mod07/Labfiles/Solution/CS/ClassTracker_CS/ClassTracker_CS/NonCreditCourse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClassTracker_CS
{
    class NonCreditCourse : Course
    {
        public char CalculateGrade(float grade)
        {
            char letterGrade;
            
            if (grade >= 90)
            {
                letterGrade = 'A';
            }
            else if (grade >= 80 && grade <= 89)
            {
                letterGrade = 'B';
            }
            else if (grade >= 70 && grade <= 79)
            {
                letterGrade = 'C';
            }
            else if (grade >= 66 && grade <= 69)
            {
                letterGrade = 'D';
            }
            else
            {
                letterGrade = 'F';
            }

            return letterGrade;
        }
    }
}
